torch/_functorch/aot_autograd.py

```diff
@@ -4135,6 +4140,19 @@ def convert(idx, x):
                     keep_input_mutations=aot_config.keep_inference_input_mutations and not needs_autograd,
                     is_train=needs_autograd,
                 )(*fake_flat_args)
+                if needs_autograd and not any(x.requires_grad for x in fw_metadata.output_info):
+                    # A bit silly: we realized that none of the outputs require grad,
+                    # so we actually have an inference graph.
+                    # Need to recompute metadata since needs_autograd determines some of our ViewAndMutationMeta
+                    # TODO: we should be able to update run_functionalized_fw_and_collect_metadata
```

Comment:
I think this will increase compilation time - is it possible to avoid?

Reply:
Yeah- very dumb but this only matters for subclasses. I’ll update it to only do this for subclasses and a TODO to clean it up